docs/cookbook/deploy-with-kamal.md

```diff
@@ -0,0 +1,233 @@
+to ensure your Rails application can connect to the correct SSR server. The value **_VITE_RUBY_HOST: "vite_ssr"_**
+must match the **_network-alias_** defined in the `vite_ssr` role above.
+
+```yml
+# Inject ENV variables into containers (secrets come from .kamal/secrets).
+env:
+  secret:
+    - RAILS_MASTER_KEY
+  clear:
+    # Run the Solid Queue Supervisor inside the web server's Puma process to do jobs.
+    # When you start using multiple servers, you should split out job processing to a dedicated machine.
+    SOLID_QUEUE_IN_PUMA: true
+
+    VITE_RUBY_HOST: "vite_ssr" // [!code ++]
+
+    # Set number of processes dedicated to Solid Queue (default: 1)
+    # JOB_CONCURRENCY: 3
+
+    # Set number of cores available to the application on each server (default: 1).
+    # WEB_CONCURRENCY: 2
+
+    # Match this to any external database server to configure Active Record correctly
+    # Use inertia_rails_svelte5_ssr-db for a db accessory server on same machine via local kamal docker network.
+    # DB_HOST: 192.168.0.2
+
+    # Log everything from Rails
+    # RAILS_LOG_LEVEL: debug
+
+```
+
+
+## Update asset_path in`config/deploy.yml`
+
+Update the asset_path to `/rails/public/vite` if you haven't.
+
+```yml
+# Bridge fingerprinted assets, like JS and CSS, between versions to avoid
+# hitting 404 on in-flight requests. Combines all files from new and old
+# version inside the asset_path.
+asset_path: /rails/public/assets // [!code --]
+asset_path: /rails/public/vite // [!code ++]
+```
+
+
+## Ensure that your `vite.config.ts` is configured to support SSR
+
+Configure Vite with an `ssr` block in your `vite.config.ts` file to ensures all dependencies are bundled for SSR.
+
+```js
+import { svelte } from '@sveltejs/vite-plugin-svelte'
+import { defineConfig } from 'vite'
+import ViteRails from "vite-plugin-rails"
+
+export default defineConfig({
+    ssr: {// [!code ++]
+        noExternal: true,// [!code ++]
+    },// [!code ++]
+    plugins: [
+        svelte(),
+        ViteRails({
+            envVars: { RAILS_ENV: "development" },
+            envOptions: { defineOn: "import.meta.env" },
+            fullReload: {
+                additionalPaths: [],
+            },
+        }),
+    ],
+})
+```
+
+## Configure SSR URL in the Inertia's Rails adapter
+
+To enable Server-Side Rendering (SSR) in your Inertia Rails application, you need to specify
+the correct SSR server URL in the adapter. By default, the adapter points to `http://localhost:13714`,
+but this must align with the **_VITE_RUBY_HOST_** value defined in your `deploy.yml` when we deploy it to production.
+
+Update the configuration in `config/initializers/inertia_rails.rb` to dynamically construct the
+SSR URL using Vite Ruby's protocol, host, and port settings:
+
+```ruby
+InertiaRails.configure do |config|
+  config.ssr_enabled = ViteRuby.config.ssr_build_enabled
+  config.ssr_url = "#{ViteRuby.config.protocol}://#{ViteRuby.config.host}:13714" # [!code ++]
```

Comment:
For clarity, the default value of `ssr_url` was added in:

- #74

`inertia_rails` does not depend on `vite_ruby`, nor is `vite_ruby` so commonly used that it would make sense to auto-detect it and use it as a default.

In addition, I would suggest to avoid using `ViteRuby.config`, as it refers to the configuration for the _Vite **Dev** Server_, and __not__ the SSR server.

If it works, it's by coincidence.

For example, a common setup consists in running the SSR server in the same box and using plain HTTP (without SSL), while in many cases it's desirable to run the Vite dev server over SSL so that HTTP2 can be used and connections are multiplexed (loading unbundled files faster in development).

Reply:
Yeah, I guess this part is a leftover from when I tried to understand what `bin/vite ssr` does (which is basically just a shortcut) 😅

I think it's better to use something like this instead: `ssr_url: ENV.fetch('INERTIA_SSR_URL', 'http://localhost:13714')` and set `INERTIA_SSR_URL` in the Kamal deployment config (any suggestions for a better name?).

Reply:
Thanks @ElMassimo, @skryukov for our detailed discussion and suggestion. I would agree with all those valid points and prepare another PR to allow to set Inertia SSR URL via the ENV variable:
* https://github.com/inertiajs/inertia-rails/pull/172